### PR TITLE
Authorization

### PR DESCRIPTION
**Pull request by InfernoJJ**
Front end authorization layer. 

Determines if the endpoint exists and what scopes to authorize to use that endpoint.

### CODE REVIEW

File: pkg/endpoints/model_validation.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+}
+
+func (c SecurityConfig) Validate() error {
+	return validation.ValidateStruct(&c,
+		validation.Field(&c.Identity, validation.Each(validation.By(validateTemplating))),
+		validation.Field(&c.Keyring, validation.Each(validation.By(validateTemplating))),
+	)
+}
+
+func validateTemplating(value interface{}) error {
```
Comment (adamdecaf) [inline pkg/endpoints/model_validation.go:74]: Do we need this custom template validation? Between `template.MustParse` and a quick `.Exec` we can rely on the standard library for verifying templates are valid.

https://github.com/moov-io/paygate/blob/master/cmd/server/main.go#L254

**Comment by InfernoJJ** [comment]
This is so it matches what open API does and it’l it’s simple.

**Comment by InfernoJJ** [comment]
We also need to reverse the template and find the values from a template so
it's bidirectional.

**Comment by adamdecaf** [comment]
Is this PR the library that apps will use to extract auth context information? I'm not quite following this.

**Comment by InfernoJJ** [comment]
No this PR is to put authorizatioon scopes to that a user/key must have to be able to use the specific API.

The library code was added awhile ago in the `/pkg/middleware`

**Pull request merged**